dts/arm/nordic/nrf5340_cpuappns.dtsi

```diff
@@ -72,6 +72,11 @@
 		compatible = "zephyr,psa-crypto-rng";
 		status = "okay";
 	};
+
+	uicr: uicr {
+		compatible = "nordic,nrf-uicr-config-only";
+		status = "okay";
+	};
 };
```

Comment:
As far as I can tell from the code here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/soc_flash_nrf.c
uicr is sub-device of flash controller. That is my first issue.
My second issue: as far as I understand "compatible" describes a device/driver the node is compatible to; the usage here is hacky.